content/shmem_team_split_strided.tex

```diff
@@ -74,20 +74,20 @@
 See Section~\ref{subsec:shmem_team_config_t} for field mask names and
 default configuration parameters.
 
-When \VAR{parent\_team} specifies an invalid team, if \VAR{parent\_team}
+If \VAR{parent\_team}
 compares equal to \LibConstRef{SHMEM\_TEAM\_INVALID}, then no new team
 will be created and \VAR{new\_team} will be assigned the value
-\LibConstRef{SHMEM\_TEAM\_INVALID}; otherwise, the behavior is undefined.
+\LibConstRef{SHMEM\_TEAM\_INVALID}.  If \VAR{parent\_team} is otherwise invalid, the behavior is undefined.
 
 If an invalid \ac{PE} triplet is provided, then the \VAR{new\_team}
 will not be created.
 
 If \VAR{new\_team} cannot be created, then it will be assigned the value
-\LibConstRef{SHMEM\_TEAM\_INVALID}.
+\LibConstRef{SHMEM\_TEAM\_INVALID} and \FUNC{shmem\_team\_split\_strided} will return a nonzero value.
```

Comment:
Related question: `new_team` will  be `SHMEM_TEAM_INVALID` on PEs that are in the parent team but not in the child team, so do those PEs return nonzero?  Seems to me that these PEs should return zero, at least if it was a successful team creation operation across the child team (assuming we have a reduction to assure equal return values across all parent PEs).  If we agree, maybe we could specify that near [line 65](https://github.com/openshmem-org/specification/pull/305/files#diff-16302be989ad41eb9353b7a8359e728cR63-R65) (or here)?

Reply:
I'm not sure we've ever really settled on a clear behavior here. I think there are two options to consider:

> If creation of the child team is successful:
> 
> 1. All PEs in the parent team are returned zero
> 2. All (and only) PEs in the child team are returned zero; PEs in the parent team, but not in the child team are returned nonzero
> 
> In either case, the `new_team` handle references a valid team for those PEs in the child team and those PEs not in the child team have `new_team == SHMEM_TEAM_INVALID`.
> 
> If creation of the child team is not successful, then all PEs in the parent team should receive a nonzero return value and  `new_team == SHMEM_TEAM_INVALID`.

I think Option 1 is the most clear and user-friendly. It's also most consistent with the return behavior discussed in your earlier question. My rationale here is that the PEs in the parent team may need to do something collectively if the child team is not created. This would be an action of the parent team, not the (nonexistent) child team, so all PEs in the parent team should know whether the team creation operation was successful.

Reply:
@davidozog What do you think of the following?

![shmem_team_split_strided](https://user-images.githubusercontent.com/2961664/68775217-25a9bd80-05fc-11ea-8810-e247586f54c4.png)

Reply:
This looks great, as long as it's also clear that an unsuccessful creation returns equal nonzero return values.